closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -543,58 +741,51 @@ JNIEXPORT jint JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_GCMDecrypt
         }
     }
 
-    if (first_time_gcm == 0) {
-        //printf("Initializing OpenSSL GCM algorithm-1\n");
-        OpenSSL_add_all_algorithms();
-        ERR_load_crypto_strings();
-        first_time_gcm = 1;
-    }
-
     switch(keyLen) {
         case 16:
-            evp_gcm_cipher = EVP_aes_128_gcm();
+            evp_gcm_cipher = (*OSSL_aes_128_gcm)();
             break;
         case 24:
-            evp_gcm_cipher = EVP_aes_192_gcm();
+            evp_gcm_cipher = (*OSSL_aes_192_gcm)();
             break;
         case 32:
-            evp_gcm_cipher = EVP_aes_256_gcm();
+            evp_gcm_cipher = (*OSSL_aes_256_gcm)();
             break;
         default:
-            break;
+            assert(32);
```

Comment:
I don't think `assert()` is the best way to handle errors; and this use is just wrong as `assert()` only treats a zero argument as a failure.